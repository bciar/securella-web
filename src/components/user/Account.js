import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper, GridList, GridTile, Avatar } from 'material-ui';

export class Account extends React.Component {
  constructor(props) {
    super(props);
    
    let image_url = Auth.user.image;
    if (!!this.props.location.state) 
        image_url = this.props.location.state.updatedImageURL;
    Auth.user.image = image_url;
  }

  componentWillMount() {
    if (!this.props.user.loggedIn) {
      this.props.history.push('/login');
    }
  }
  
  render() {
    return (
        <div className="container">
            <Paper className="account-paper" zDepth={4}>
                <div id='root'>
                    <div id="first_list">
                      <GridList cols={1} cellHeight={80}>
                          <GridTile rows={1} size={70} className="grid-title">
                            <Avatar size={70} src={ Auth.user.image}  />
                          </GridTile>
                          <GridTile size={30} className="grid-value">
                            <RaisedButton href="/update_image" label="change" />
                          </GridTile>
                      </GridList>
                    </div>
                    <GridList cellHeight={40}>
                        <GridTile className="grid-title"> Name: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.name} </GridTile>
                        <GridTile className="grid-title"> Adresse: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.address} </GridTile>
                        <GridTile className="grid-title"> </GridTile>
                        <GridTile className="grid-value"> {Auth.user.postal_code} {Auth.user.city} </GridTile>
                        <GridTile className="grid-title"> Gender: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.gender} </GridTile>
                        <GridTile className="grid-title"> Age: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.age} </GridTile>
                        <RaisedButton href="/update_profile" label="Update profile" />
                        <RaisedButton href="/reset_password" label="Change password" />
                    </GridList>
                </div>
            </Paper>
        </div>
      );
    }
}
