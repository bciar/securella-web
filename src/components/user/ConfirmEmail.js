import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

export class ConfirmEmail extends Component {

    constructor(props) {
        super(props);
        this.state = {
            confirmed: true,
        };
    }

    render() {
      if (this.state.confirmed) {
        return (
          <div className="container">
            <Paper className="common-paper" zDepth={1}>
              Thank you for confirming your email. You may now proceed to the
              <br/>
              <b><Link to='/login'>Login</Link></b>
            </Paper>
          </div>
        );
      }
      else {
        return (
          <div className="container">
            <Paper className="common-paper" zDepth={1}>
              Sorry, we could not confirm your account.
              <br/>
              <b>Please call 911 for help.</b>
            </Paper>
          </div>
        );
      }
    }
}
