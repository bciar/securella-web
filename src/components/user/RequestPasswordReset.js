import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';

export class RequestPasswordReset extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            formData: {
                email: '',
                password: ''
            },
            submitted: false,
            errors: false
        };
    }

    componentWillMount() {
      Auth.validateToken()
        .then(function(user) {
        this.props.history.push('/reset_password');
      });
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        const { email } = this.state.formData;
        this.setState({ submitted: true }, () => {
          Auth.requestPasswordReset({
            email,
            redirect_url: '/reset_password'
          })
          .then(function(user) {
            history.push('/request_password_success');
          })
          .fail(res => {
            document.getElementById("errorLabel").className = '';
            this.setState({ submitted: false });
          })
        });
    }

    render() {
        const { formData, submitted } = this.state;
        if (Auth.user.signedIn) {
            return (<div><h1>You are in!</h1></div>);
        }
        else {
            return (
                <div className="container">
                    <Paper className="common-paper">
                        <ValidatorForm
                            ref="form"
                            onSubmit={ () => this.handleSubmit(this.props.history) } >
                            <h2>Reset your Password</h2>
                            <h3 id="errorLabel" className="hidden"> Email or password wrong</h3>
                            <TextValidator
                                floatingLabelText="Email"
                                onChange={this.handleChange}
                                name="email"
                                id="email"
                                value={formData.email}
                                validators={['required']}
                                errorMessages={['this field is required']}
                            />
                            <br />
                            <RaisedButton
                                type="submit"
                                label={
                                    (submitted && 'Processing...')
                                    || (!submitted && 'Get a new password')
                                }
                                disabled={submitted}
                                style={{marginTop:'50px', marginBottom:'10px'}}
                            />
                        </ValidatorForm>
                    </Paper>
                </div>
            );
        }
    }
}
