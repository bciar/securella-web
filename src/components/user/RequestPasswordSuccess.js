import React from 'react';
import Paper from 'material-ui/Paper';

export const RequestPasswordSuccess = () => (
  <div className="container">
    <Paper className="common-paper" zDepth={1}>
      If your Email is registered, you will receive an email with a link to change your password within the next 15 minutes.
      <br/>
      <b>Please check your inbox.</b>
    </Paper>
  </div>
);