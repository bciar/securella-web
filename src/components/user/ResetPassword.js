import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';


export class ResetPassword extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formData: {
                password: '',
                repeatPassword: '',
            },
            submitted: false,
            errors: '0',
        };
    }

    componentWillMount() {
        ValidatorForm.addValidationRule('isPasswordMatch', (value) => {
            if (value !== this.state.formData.password) {
                return false;
            }
            return true;
        });
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        const { password, repeatPassword } = this.state.formData;
        this.setState({ submitted: true }, () => {
          Auth.updatePassword({
            password: password,
            password_confirmation: repeatPassword
          })
          .then(function() {
            history.push('/reset_password_success');
          })
          .fail(res => {
            document.getElementById("errorLabel").className = '';
            this.setState({ submitted: false });
          })
        });
    }

    render() {
        const { formData, submitted } = this.state;
        return (
            <div className="container">
                <Paper className="common-paper">
                    <ValidatorForm
                        ref="form"
                        onSubmit={ () => this.handleSubmit(this.props.history) } >
                        <h2>Passwort updaten</h2>
                        <h3 id="errorLabel" className="hidden"> Password zu kurz (min. 8 Zeichen)</h3>
                        <TextValidator
                            floatingLabelText="Password"
                            onChange={this.handleChange}
                            name="password"
                            type="password"
                            id="password"
                            validators={['required']}
                            errorMessages={['this field is required']}
                            value={formData.password}
                        />
                        <br />
                        <TextValidator
                            floatingLabelText="Repeat password"
                            onChange={this.handleChange}
                            name="repeatPassword"
                            type="password"
                            id="password_confirmation"
                            validators={['isPasswordMatch', 'required']}
                            errorMessages={['password mismatch', 'this field is required']}
                            value={formData.repeatPassword}
                        />
                        <br />
                        <RaisedButton
                            type="submit"
                            label={
                                (submitted && 'Updating your password!')
                                || (!submitted && 'Passwort updaten')
                            }
                            disabled={submitted}
                            style={{marginTop:'50px', marginBottom:'10px'}}
                        />
                    </ValidatorForm>
                </Paper>
            </div>
        );

    }
}
