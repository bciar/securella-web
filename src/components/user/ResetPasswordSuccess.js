import React from 'react';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

export const ResetPasswordSuccess = () => (
  <div className="container">
    <Paper className="common-paper" zDepth={1}>
      <h1>You are updated your password</h1>
      <p>
        <Link to='/login'>Login</Link>&nbsp;&nbsp;here now.
      </p>
    </Paper>
  </div>
);