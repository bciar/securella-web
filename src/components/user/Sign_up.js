import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';
import { Redirect } from 'react-router-dom';


export class SignUp extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            formData: {
                email: '',
                password: '',
                repeatPassword: '',
            },
            submitted: false,
            errors: '0',
        };
    }

    componentWillMount() {
        ValidatorForm.addValidationRule('isPasswordMatch', (value) => {
            if (value !== this.state.formData.password) {
                return false;
            }
            return true;
        });
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        this.setState({ submitted: true }, () => {
          Auth.emailSignUp({
            email: this.state.email, //document.getElementById('email').value,
            password: this.state.password, //document.getElementById('password').value,
            password_confirmation: this.state.repeatPassword//document.getElementById('password_confirmation').value
          })
          .then(function() {
            history.push('/verify');
          })
          .fail(res => {
            document.getElementById("errorLabel").className = '';
            this.setState({ submitted: false });
          })
        });
    }

    renderSpinner = () => {
      return ('Loading...')
    }

    renderApp() {
        const { formData, submitted } = this.state;
        if (this.props.user.loggedIn) { return <Redirect to='/dashboard' />; }
        else {
            return (
                <div className="container">
                    <Paper className="common-paper">
                    <ValidatorForm
                        ref="form"
                        onSubmit={ () => this.handleSubmit(this.props.history) } >
                        <h2>Create your Account</h2>
                        <h3 id="errorLabel" className="hidden"> Email vergeben oder Password zu kurz (min. 8 Zeichen)</h3>
                        <TextValidator
                            floatingLabelText="Email"
                            onChange={this.handleChange}
                            name="email"
                            id="email"
                            value={formData.email}
                            validators={['required', 'isEmail']}
                            errorMessages={['this field is required', 'email is not valid']}
                        />
                        <br />
                        <TextValidator
                            floatingLabelText="Password"
                            onChange={this.handleChange}
                            name="password"
                            type="password"
                            id="password"
                            validators={['required']}
                            errorMessages={['this field is required']}
                            value={formData.password}
                        />
                        <br />
                        <TextValidator
                            floatingLabelText="Repeat password"
                            onChange={this.handleChange}
                            name="repeatPassword"
                            type="password"
                            id="password_confirmation"
                            validators={['isPasswordMatch', 'required']}
                            errorMessages={['password mismatch', 'this field is required']}
                            value={formData.repeatPassword}
                        />
                        <br />
                        <RaisedButton
                            type="submit"
                            label={
                                (submitted && 'Your account is being created!')
                                || (!submitted && 'Account alegen')
                            }
                            disabled={submitted}
                            style={{marginTop:'50px', marginBottom:'10px'}}
                        />
                    </ValidatorForm>
                    </Paper>
                </div>
            );
        }
    }

    render() {
        return this.props.app.loaded ? this.renderApp() : ('Loading...');
    }
}
