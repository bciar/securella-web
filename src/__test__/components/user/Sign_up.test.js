import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from '../../../components/user/Sign_up';

let user, wrapper;
const signup_test_data = {
    email: 'test-user@mail.com', 
    password: 'password', 
    repeatPassword: 'password'
};

beforeEach(() => {
    user = jest.fn();
    wrapper = shallow(<SignUp app={{loaded: true}} user={user} />);
});

test('should render Signup page correctly when the loaded state is true', () => {
    expect(wrapper).toMatchSnapshot();
});

test('spinner should be rendered if the app loaded state is false', () => {
    wrapper = shallow(<SignUp app={{loaded: false}} user={user} />);
    expect(wrapper).toMatchSnapshot();
});

test('formData state should be changed along the text are inputted', () => {
    wrapper.find('TextValidator').at(0).simulate('change', {
        target: { name: 'email', value: signup_test_data.email }
    });
    wrapper.find('TextValidator').at(1).simulate('change', {
        target: { name: 'password', value: signup_test_data.password }
    });
    wrapper.find('TextValidator').at(2).simulate('change', {
        target: { name: 'repeatPassword', value: signup_test_data.repeatPassword }
    });
    expect(wrapper.state().formData).toEqual(signup_test_data);
});

test('submit form should work correctly', () => {
    wrapper.setState({
        formData: signup_test_data
    });
    const history = jest.fn();
    expect(wrapper.find('TextValidator').at(0).props().value).toEqual(signup_test_data.email);
    expect(wrapper.find('TextValidator').at(1).props().value).toEqual(signup_test_data.password);
    expect(wrapper.find('TextValidator').at(2).props().value).toEqual(signup_test_data.repeatPassword);
    wrapper.find('ValidatorForm').simulate('submit', history);
    expect(wrapper.state().submitted).toEqual(true);
});
