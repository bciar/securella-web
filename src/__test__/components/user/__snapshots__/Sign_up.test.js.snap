// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Signup page correctly when the loaded state is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    app={
        Object {
            "loaded": true,
          }
    }
    user={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Paper
        circle={false}
        className="common-paper"
        rounded={true}
        transitionEnabled={true}
        zDepth={1}
>
        <ValidatorForm
                onError={[Function]}
                onSubmit={[Function]}
        >
                <h2>
                        Create your Account
                </h2>
                <h3
                        className="hidden"
                        id="errorLabel"
                >
                         Email vergeben oder Password zu kurz (min. 8 Zeichen)
                </h3>
                <TextValidator
                        errorMessages={
                                Array [
                                        "this field is required",
                                        "email is not valid",
                                      ]
                        }
                        floatingLabelText="Email"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        validatorListener={[Function]}
                        validators={
                                Array [
                                        "required",
                                        "isEmail",
                                      ]
                        }
                        value=""
                />
                <br />
                <TextValidator
                        errorMessages={
                                Array [
                                        "this field is required",
                                      ]
                        }
                        floatingLabelText="Password"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        validatorListener={[Function]}
                        validators={
                                Array [
                                        "required",
                                      ]
                        }
                        value=""
                />
                <br />
                <TextValidator
                        errorMessages={
                                Array [
                                        "password mismatch",
                                        "this field is required",
                                      ]
                        }
                        floatingLabelText="Repeat password"
                        id="password_confirmation"
                        name="repeatPassword"
                        onChange={[Function]}
                        type="password"
                        validatorListener={[Function]}
                        validators={
                                Array [
                                        "isPasswordMatch",
                                        "required",
                                      ]
                        }
                        value=""
                />
                <br />
                <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Account alegen"
                        labelPosition="after"
                        primary={false}
                        secondary={false}
                        style={
                                Object {
                                        "marginBottom": "10px",
                                        "marginTop": "50px",
                                      }
                        }
                        type="submit"
                />
        </ValidatorForm>
</Paper>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ValidatorForm
          onError={[Function]}
          onSubmit={[Function]}
>
          <h2>
                    Create your Account
          </h2>
          <h3
                    className="hidden"
                    id="errorLabel"
          >
                     Email vergeben oder Password zu kurz (min. 8 Zeichen)
          </h3>
          <TextValidator
                    errorMessages={
                              Array [
                                        "this field is required",
                                        "email is not valid",
                                      ]
                    }
                    floatingLabelText="Email"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    validatorListener={[Function]}
                    validators={
                              Array [
                                        "required",
                                        "isEmail",
                                      ]
                    }
                    value=""
          />
          <br />
          <TextValidator
                    errorMessages={
                              Array [
                                        "this field is required",
                                      ]
                    }
                    floatingLabelText="Password"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    type="password"
                    validatorListener={[Function]}
                    validators={
                              Array [
                                        "required",
                                      ]
                    }
                    value=""
          />
          <br />
          <TextValidator
                    errorMessages={
                              Array [
                                        "password mismatch",
                                        "this field is required",
                                      ]
                    }
                    floatingLabelText="Repeat password"
                    id="password_confirmation"
                    name="repeatPassword"
                    onChange={[Function]}
                    type="password"
                    validatorListener={[Function]}
                    validators={
                              Array [
                                        "isPasswordMatch",
                                        "required",
                                      ]
                    }
                    value=""
          />
          <br />
          <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="Account alegen"
                    labelPosition="after"
                    primary={false}
                    secondary={false}
                    style={
                              Object {
                                        "marginBottom": "10px",
                                        "marginTop": "50px",
                                      }
                    }
                    type="submit"
          />
</ValidatorForm>,
        "circle": false,
        "className": "common-paper",
        "rounded": true,
        "transitionEnabled": true,
        "zDepth": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h2>
              Create your Account
</h2>,
            <h3
              className="hidden"
              id="errorLabel"
>
               Email vergeben oder Password zu kurz (min. 8 Zeichen)
</h3>,
            <TextValidator
              errorMessages={
                            Array [
                                          "this field is required",
                                          "email is not valid",
                                        ]
              }
              floatingLabelText="Email"
              id="email"
              name="email"
              onChange={[Function]}
              validatorListener={[Function]}
              validators={
                            Array [
                                          "required",
                                          "isEmail",
                                        ]
              }
              value=""
/>,
            <br />,
            <TextValidator
              errorMessages={
                            Array [
                                          "this field is required",
                                        ]
              }
              floatingLabelText="Password"
              id="password"
              name="password"
              onChange={[Function]}
              type="password"
              validatorListener={[Function]}
              validators={
                            Array [
                                          "required",
                                        ]
              }
              value=""
/>,
            <br />,
            <TextValidator
              errorMessages={
                            Array [
                                          "password mismatch",
                                          "this field is required",
                                        ]
              }
              floatingLabelText="Repeat password"
              id="password_confirmation"
              name="repeatPassword"
              onChange={[Function]}
              type="password"
              validatorListener={[Function]}
              validators={
                            Array [
                                          "isPasswordMatch",
                                          "required",
                                        ]
              }
              value=""
/>,
            <br />,
            <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Account alegen"
              labelPosition="after"
              primary={false}
              secondary={false}
              style={
                            Object {
                                          "marginBottom": "10px",
                                          "marginTop": "50px",
                                        }
              }
              type="submit"
/>,
          ],
          "onError": [Function],
          "onSubmit": [Function],
        },
        "ref": "form",
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create your Account",
            },
            "ref": null,
            "rendered": "Create your Account",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Email vergeben oder Password zu kurz (min. 8 Zeichen)",
              "className": "hidden",
              "id": "errorLabel",
            },
            "ref": null,
            "rendered": " Email vergeben oder Password zu kurz (min. 8 Zeichen)",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errorMessages": Array [
                "this field is required",
                "email is not valid",
              ],
              "floatingLabelText": "Email",
              "id": "email",
              "name": "email",
              "onChange": [Function],
              "validatorListener": [Function],
              "validators": Array [
                "required",
                "isEmail",
              ],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errorMessages": Array [
                "this field is required",
              ],
              "floatingLabelText": "Password",
              "id": "password",
              "name": "password",
              "onChange": [Function],
              "type": "password",
              "validatorListener": [Function],
              "validators": Array [
                "required",
              ],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errorMessages": Array [
                "password mismatch",
                "this field is required",
              ],
              "floatingLabelText": "Repeat password",
              "id": "password_confirmation",
              "name": "repeatPassword",
              "onChange": [Function],
              "type": "password",
              "validatorListener": [Function],
              "validators": Array [
                "isPasswordMatch",
                "required",
              ],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "fullWidth": false,
              "label": "Account alegen",
              "labelPosition": "after",
              "primary": false,
              "secondary": false,
              "style": Object {
                "marginBottom": "10px",
                "marginTop": "50px",
              },
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Paper
          circle={false}
          className="common-paper"
          rounded={true}
          transitionEnabled={true}
          zDepth={1}
>
          <ValidatorForm
                    onError={[Function]}
                    onSubmit={[Function]}
          >
                    <h2>
                              Create your Account
                    </h2>
                    <h3
                              className="hidden"
                              id="errorLabel"
                    >
                               Email vergeben oder Password zu kurz (min. 8 Zeichen)
                    </h3>
                    <TextValidator
                              errorMessages={
                                        Array [
                                                  "this field is required",
                                                  "email is not valid",
                                                ]
                              }
                              floatingLabelText="Email"
                              id="email"
                              name="email"
                              onChange={[Function]}
                              validatorListener={[Function]}
                              validators={
                                        Array [
                                                  "required",
                                                  "isEmail",
                                                ]
                              }
                              value=""
                    />
                    <br />
                    <TextValidator
                              errorMessages={
                                        Array [
                                                  "this field is required",
                                                ]
                              }
                              floatingLabelText="Password"
                              id="password"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              validatorListener={[Function]}
                              validators={
                                        Array [
                                                  "required",
                                                ]
                              }
                              value=""
                    />
                    <br />
                    <TextValidator
                              errorMessages={
                                        Array [
                                                  "password mismatch",
                                                  "this field is required",
                                                ]
                              }
                              floatingLabelText="Repeat password"
                              id="password_confirmation"
                              name="repeatPassword"
                              onChange={[Function]}
                              type="password"
                              validatorListener={[Function]}
                              validators={
                                        Array [
                                                  "isPasswordMatch",
                                                  "required",
                                                ]
                              }
                              value=""
                    />
                    <br />
                    <RaisedButton
                              disabled={false}
                              fullWidth={false}
                              label="Account alegen"
                              labelPosition="after"
                              primary={false}
                              secondary={false}
                              style={
                                        Object {
                                                  "marginBottom": "10px",
                                                  "marginTop": "50px",
                                                }
                              }
                              type="submit"
                    />
          </ValidatorForm>
</Paper>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ValidatorForm
            onError={[Function]}
            onSubmit={[Function]}
>
            <h2>
                        Create your Account
            </h2>
            <h3
                        className="hidden"
                        id="errorLabel"
            >
                         Email vergeben oder Password zu kurz (min. 8 Zeichen)
            </h3>
            <TextValidator
                        errorMessages={
                                    Array [
                                                "this field is required",
                                                "email is not valid",
                                              ]
                        }
                        floatingLabelText="Email"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        validatorListener={[Function]}
                        validators={
                                    Array [
                                                "required",
                                                "isEmail",
                                              ]
                        }
                        value=""
            />
            <br />
            <TextValidator
                        errorMessages={
                                    Array [
                                                "this field is required",
                                              ]
                        }
                        floatingLabelText="Password"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        type="password"
                        validatorListener={[Function]}
                        validators={
                                    Array [
                                                "required",
                                              ]
                        }
                        value=""
            />
            <br />
            <TextValidator
                        errorMessages={
                                    Array [
                                                "password mismatch",
                                                "this field is required",
                                              ]
                        }
                        floatingLabelText="Repeat password"
                        id="password_confirmation"
                        name="repeatPassword"
                        onChange={[Function]}
                        type="password"
                        validatorListener={[Function]}
                        validators={
                                    Array [
                                                "isPasswordMatch",
                                                "required",
                                              ]
                        }
                        value=""
            />
            <br />
            <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Account alegen"
                        labelPosition="after"
                        primary={false}
                        secondary={false}
                        style={
                                    Object {
                                                "marginBottom": "10px",
                                                "marginTop": "50px",
                                              }
                        }
                        type="submit"
            />
</ValidatorForm>,
          "circle": false,
          "className": "common-paper",
          "rounded": true,
          "transitionEnabled": true,
          "zDepth": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h2>
                Create your Account
</h2>,
              <h3
                className="hidden"
                id="errorLabel"
>
                 Email vergeben oder Password zu kurz (min. 8 Zeichen)
</h3>,
              <TextValidator
                errorMessages={
                                Array [
                                                "this field is required",
                                                "email is not valid",
                                              ]
                }
                floatingLabelText="Email"
                id="email"
                name="email"
                onChange={[Function]}
                validatorListener={[Function]}
                validators={
                                Array [
                                                "required",
                                                "isEmail",
                                              ]
                }
                value=""
/>,
              <br />,
              <TextValidator
                errorMessages={
                                Array [
                                                "this field is required",
                                              ]
                }
                floatingLabelText="Password"
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
                validatorListener={[Function]}
                validators={
                                Array [
                                                "required",
                                              ]
                }
                value=""
/>,
              <br />,
              <TextValidator
                errorMessages={
                                Array [
                                                "password mismatch",
                                                "this field is required",
                                              ]
                }
                floatingLabelText="Repeat password"
                id="password_confirmation"
                name="repeatPassword"
                onChange={[Function]}
                type="password"
                validatorListener={[Function]}
                validators={
                                Array [
                                                "isPasswordMatch",
                                                "required",
                                              ]
                }
                value=""
/>,
              <br />,
              <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Account alegen"
                labelPosition="after"
                primary={false}
                secondary={false}
                style={
                                Object {
                                                "marginBottom": "10px",
                                                "marginTop": "50px",
                                              }
                }
                type="submit"
/>,
            ],
            "onError": [Function],
            "onSubmit": [Function],
          },
          "ref": "form",
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create your Account",
              },
              "ref": null,
              "rendered": "Create your Account",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Email vergeben oder Password zu kurz (min. 8 Zeichen)",
                "className": "hidden",
                "id": "errorLabel",
              },
              "ref": null,
              "rendered": " Email vergeben oder Password zu kurz (min. 8 Zeichen)",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": Array [
                  "this field is required",
                  "email is not valid",
                ],
                "floatingLabelText": "Email",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "validatorListener": [Function],
                "validators": Array [
                  "required",
                  "isEmail",
                ],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": Array [
                  "this field is required",
                ],
                "floatingLabelText": "Password",
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "validatorListener": [Function],
                "validators": Array [
                  "required",
                ],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": Array [
                  "password mismatch",
                  "this field is required",
                ],
                "floatingLabelText": "Repeat password",
                "id": "password_confirmation",
                "name": "repeatPassword",
                "onChange": [Function],
                "type": "password",
                "validatorListener": [Function],
                "validators": Array [
                  "isPasswordMatch",
                  "required",
                ],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Account alegen",
                "labelPosition": "after",
                "primary": false,
                "secondary": false,
                "style": Object {
                  "marginBottom": "10px",
                  "marginTop": "50px",
                },
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`spinner should be rendered if the app loaded state is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    app={
        Object {
            "loaded": false,
          }
    }
    user={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Loading...",
  Symbol(enzyme.__nodes__): Array [
    "Loading...",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
